cob_people_detection/ros/src/face_recognizer_node.cpp

```diff
@@ -518,7 +518,7 @@ void FaceRecognizerNode::facePositionsCallback(const cob_perception_msgs::ColorD
 				// set label
 				det.label = identification_labels[head][face];
 				// set origin of detection
-				det.detector = "face";
+				det.detector = ros::this_node::getName();
```

Comment:
label is the name of the identified person, I have no idea about id, is it set by people detection? The message format is inherited from object detection, so id might be free

However, detector is the field that is intended to be used as a clue to the source. What does ros::this_node::getName() yield? You may also put into the detector field: "head," + ros::this_node::getName() and later adapt the code in the detection tracker to search whether detector contains/begins with "head" or "face" and still use the remaining information of the string elsewhere.